lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+                  , [ expectListSizeEqual 2
+                    , expectListItemFieldEqual 0 amount a2
+                    , expectListItemFieldEqual 1 amount a1
+                    ]
+                  )
+                -- 16
+                , ( "?end="
+                    <> (toParam $ addUTCTime (-1) t2)
+                  , [ expectListSizeEqual 1
+                    , expectListItemFieldEqual 0 amount a1
+                    ]
+                  )
+                ]
+
+        forM_ matrix $ \(query, expectations) -> do
+          rf <- request @([ApiTransaction t]) ctx (listTxEp w query)
+              Default Empty
+          verify rf expectations
+
+    describe "TRANS_LIST_02,03 - Faulty start, end, order values" $ do
+        let orderErr = "Please specify one of the following values:\
+            \ ascending, descending."
+        let startEndErr = "Expecting ISO 8601 date-and-time format\
+            \ (basic or extended), e.g. 2012-09-25T10:15:00Z."
+        let queries =
+                  [ ( "?start=2009"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage startEndErr
+                      ]
+                    )
+                  ,
+                    ( "?start=2012-09-25T10:15:00Z&end=2016-11-21"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage startEndErr
+                      ]
+                    )
+                  ,
+                    ( "?start=2012-09-25&start=2016-11-21T10:15:00Z"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage startEndErr
+                      ]
+                    )
+                  ,
+                    ( "?end=2012-09-25T10:15:00Z&start=2016-11-21"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage startEndErr
+                      ]
+                    )
+                  ,
+                    ( "?order=scending"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage orderErr
+                      ]
+                    )
+                  ,
+                    ( "?start=2012-09-25T10:15:00Z&order=asc"
+                    , [ expectResponseCode @IO HTTP.status400
+                      , expectErrorMessage orderErr
+                      ]
+                    )
+                  ]
+        forM_ queries $ \(q, expects) -> it q $ \ctx -> do
+            w <- emptyWallet ctx
+            let query = T.pack q
+            r <- request @([ApiTransaction t]) ctx (listTxEp w query)
+                Default Empty
+            verify r expects
+
     it "TRANS_LIST_02 - Start time shouldn't be later than end time" $
         \ctx -> do
               w <- emptyWallet ctx
               let startTime = "2009-09-09T09:09:09Z"
               let endTime = "2001-01-01T01:01:01Z"
```

Comment:
Perhaps it would also be worth adding a case that tests the inequality a bit more thoroughly.

After all, we do actually allow `start == end`.

Reply:
> After all, we do actually allow start == end.

This case is missing indeed.